packages/jest-create-cache-key-function/src/index.ts

```diff
@@ -0,0 +1,66 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ */
+
+import {createHash} from 'crypto';
+import {relative} from 'path';
+/* eslint-disable-next-line no-restricted-imports */
+import {readFileSync} from 'fs';
+import type {Config} from '@jest/types';
+
+type CacheKeyOptions = {
+  config: Config.ProjectConfig;
+  instrument: boolean;
+};
+
+type GetCacheKeyFunction = (
+  fileData: string,
+  filePath: Config.Path,
+  configStr: string,
+  options: CacheKeyOptions,
+) => string;
+
+function getGlobalCacheKey(files: Array<string>, values: Array<string>) {
+  return [
+    process.env.NODE_ENV,
+    process.env.BABEL_ENV,
+    ...values,
+    ...files.map((file: string) => readFileSync(file)),
+  ]
+    .reduce(
+      (hash, chunk) => hash.update('\0', 'utf-8').update(chunk || ''),
+      createHash('md5'),
+    )
+    .digest('hex');
+}
+
+function getCacheKeyFunction(globalCacheKey: string) {
+  return (
+    src: string,
+    file: Config.Path,
+    _configString: string,
+    options: CacheKeyOptions,
+  ) => {
+    const {config, instrument} = options;
+    const rootDir = config && config.rootDir;
```

Comment:
```suggestion
    const {rootDir} = config;
```

`config` is always defined, no?

Reply:
If not, `config?.rootDir` is cleaner and #2020